Set the right fields to allow order refund

thisiscab opened the pull request:
That way we will be able to refund the orders from Shopify directly on
Solidus

thisiscab committed:
Set the right fields to allow order refund

That way we will be able to refund the orders from Shopify directly on
Solidus

thisiscab commented:
Thumb up please? @bryanmtl @braidn @DanielWright 
Thanks!

thisiscab committed:
Stub the return values of shopify order transactions

Normally it makes another call to the API in order to fetch the
transactions of the order, but we stub it in order to not ping the API
in our test.

braidn commented on spec/models/spree/retail/shopify/generate_pos_order_spec.rb:
```diff
@@ -12,8 +12,11 @@
 
   let!(:response_mock) { mock_request('orders/450789469', 'json') }
   let(:order_response) { ShopifyAPI::Order.find('450789469') }
+  let(:shopify_transaction) { double('transaction', id: 123) }
```
Is there a way to grab this from the order information returned from Shopify's API?

thisiscab replied:
Alright, I'll take a proper look!

braidn replied:
One doesn't half-do mocks.

thisiscab replied:
You're right, is that better?

braidn reviewed (commented):
Just a small question!

thisiscab committed:
Merge branch 'master' into feature/shopify/allow-for-refund

thisiscab committed:
Properly mock the transaction requests

Instead of forcing a certain value, we properly mock the returned value
with the real json that would be returned from Shopify.

braidn commented on spec/support/shared/shopify_request.rb:
```diff
@@ -14,18 +14,17 @@
   end
 
-  def mock_request(endpoint, extension)
+  def mock_request(file, endpoint, extension)
```
Is this a file or an endpoint? Are we mocking files or endpoint connection?

thisiscab replied:
How would you go about the case we're facing where the endpoint is `orders/{order_id}/transactions`

thisiscab replied:
How would you know which files to read in that situation?

braidn replied:
maybe we check if it's an array or not? What's transactions in the case of above?

thisiscab replied:
Let's open a new PR for this if need be, I need to merge this.

thisiscab replied:
Transactions is part of the endpoint. It will reach all the transactions associated with the order, which we required in order to successfully create a refundable order.

I don't understand your `array or not` argument. 

Also github is drunk, the order of those comments don't make any sense.

braidn replied:
The order looks fine for me. If you need to merge it, that's fine.

braidn commented on spec/support/shared/shopify_request.rb:
```diff
@@ -14,18 +14,17 @@
+  def mock_request(file, endpoint, extension)
     url = "#{ShopifyAPI::Base.site}/#{endpoint}.#{extension}"
-    json = read_file_from_endpoint(endpoint, extension)
+    json = read_file(file, extension)
```
This is implementation (somewhat private) and this change is cool. Calling the mock request will use this underlying item and it will read from a file. All good here 👍

braidn reviewed (commented):
Just another small question
